target/wddr100/src/impl.c

```diff
@@ -527,6 +577,20 @@ wddr_return_t wddr_send_packets(wddr_handle_t wddr, const List_t *packets)
     return WDDR_SUCCESS;
 }
 
+__attribute__(( weak ))
+wddr_return_t wddr_unload_packets(wddr_handle_t wddr, List_t *packets)
+{
+    // Can't call unload multiple times without calling load
+    if (listLIST_IS_EMPTY(&loaded_packet_list))
+    {
+        return WDDR_ERROR;
+    }
+
+    // Transfer packets over to internal list
+    vListTransferItems(packets, &loaded_packet_list);
```

Comment:
I'm not sure if returning the previous packets is the best way to handle this, or if clearing out the list and packets and just leaving that list empty is better.  Is there time savings that is done with this, or will we probably end up clearing and rebuilding the packet list anyway?
This is also different behavior than what WDDR101 will do.

Reply:
1. If you read the API, it says that it's undefined behavior to use packets parameter after calling load but before calling unload. In both cases, it's not safe to use packets list again until you call unload. But after calling unload, in both cases packets list will be in the same state.
2. By directly clearing out list and packets in unload, we are limiting our API. We're forcing callers to not be able to use packets again after they're loaded. But there's no reason for that. They might want to modify only a single in the packet and then reload them. It doesn't make sense to reconstitute the majority of packets that weren't modified. Instead, I just gave back "ownership" of the packets to caller so that they can do whatever they like because at that point it doesn't matter to the implementation. 
2a. One other complication with freeing the packet is that we don't know if packet was preallocated or malloc'd. We can assume based on if storage parameter is used then it was preallocated but what if that implementation was to change? Now WDDR implementation would need to know how to handle it as well. Instead, I'd rather just leave that as the responsibility of the caller. 

Feel free to give me a call if you want to discuss further.

Reply:
Also, you are raising a good point. I'm not dismissing it just trying to convey my thought process when I implemented this.